src/main/java/io/appform/functionmetrics/FunctionTimerAspect.java

```diff
@@ -43,8 +52,35 @@ public Object around(ProceedingJoinPoint joinPoint) throws Throwable {
         final String methodName = Strings.isNullOrEmpty(monitoredFunction.method())
                                     ? callSignature.getName()
                                     : monitoredFunction.method();
-        log.trace("Called for class: {} method: {}", className, methodName);
-        final FunctionInvocation invocation = new FunctionInvocation(className, methodName);
+        final String[] parameterNames = methodSignature.getParameterNames();
+        String parameterString = "";
+        List<String> paramValues = Streams.zip(Arrays.stream(methodSignature.getMethod().getParameters()), Arrays.stream(joinPoint.getArgs()), Pair::new)
+                .filter(pair -> {
+                    MetricTerm metricTerm = pair.getKey().getAnnotation(MetricTerm.class);
+                    return metricTerm != null;
+                })
+                .map(pair -> {
+                    MetricTerm metricTerm = pair.getKey().getAnnotation(MetricTerm.class);
+                    return new Pair<>(metricTerm.order(), pair.getValue());
+                })
```

Comment:
Merge the two into one. Return null if not annotation and then filter on Objects::nonNull. That way you call the getAnnotation method only once.

Reply:
Created a single map lambda function merging both.